modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/gadgetLoadOrderingTest.java

```diff
@@ -0,0 +1,169 @@
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+
+import java.util.Date;
+
+/**
+ * Test load ordering for gadgets with various priorities
+ */
+public class gadgetLoadOrderingTest extends DSUIIntegrationTest {
+
+    private static final String DASHBOARD_TITLE = "sampledashboard1";
+
+    private static final String UESDASHBOARDS_GADGETSTORE_PATH = "/store" + File.separator + "carbon.super" + File.separator + "fs" + File.separator + "gadget";
+
+    /**
+     * Initializes the class.
+     *
+     * @param userMode       user mode
+     * @param dashboardTitle title of the dashboard
+     */
+    @Factory(dataProvider = "userMode")
+    public gadgetLoadOrderingTest(TestUserMode userMode, String dashboardTitle) {
+        super(userMode);
+    }
+
+    /**
+     * Provides user modes.
+     *
+     * @return
+     */
+    @DataProvider(name = "userMode")
+    public static Object[][] userModeProvider() {
+        return new Object[][]{{TestUserMode.SUPER_TENANT_ADMIN, DASHBOARD_TITLE}};
+    }
+
+    /**
+     * Setup the testing environment.
+     *
+     * @throws XPathExpressionException
+     * @throws IOException
+     * @throws AutomationUtilException
+     */
+    @BeforeClass(alwaysRun = true)
+    public void setUp() throws AutomationUtilException, XPathExpressionException, IOException {
+        String systemResourceLocation = FrameworkPathUtil.getSystemResourceLocation();
+        String gadgetStorePath = getPortalFilePath(UESDASHBOARDS_GADGETSTORE_PATH);
+        String testGadgetPathLowPriority = systemResourceLocation + "gadgets/textbox-testGadgetLowPriority.zip";
+        String testGadgetPathMediumPriority = systemResourceLocation + "gadgets/textbox-testGadgetMediumPriority.zip";
+        String testGadgetPathHighPriority = systemResourceLocation + "gadgets/textbox-testGadgetHighPriority.zip";
+
+        FileUtils.copyFileToDirectory(new File(testGadgetPathLowPriority), new File(gadgetStorePath));
+        FileUtils.copyFileToDirectory(new File(testGadgetPathMediumPriority), new File(gadgetStorePath));
+        FileUtils.copyFileToDirectory(new File(testGadgetPathHighPriority), new File(gadgetStorePath));
+
+        // Restart the server after the custom gadgets are loaded
+        AutomationContext automationContext =
+                new AutomationContext(DSIntegrationTestConstants.DS_PRODUCT_NAME, this.userMode);
+        ServerConfigurationManager serverConfigurationManager = new ServerConfigurationManager(automationContext);
+        serverConfigurationManager.restartGracefully();
+
+        login(getCurrentUsername(), getCurrentPassword());
+        addDashBoard(DASHBOARD_TITLE, "This is a test dashboard");
+    }
+
+    /**
+     * Get the full path to a file within the portal.
+     *
+     * @param relativePath Relative path to the file within a portal
+     * @return Full qualified path
+     */
+    private String getPortalFilePath(String relativePath) {
```

Comment:
Move this method DSUIIntegrationTest class in the test-ui-utils. The same method have been used in the CarbonServerAddressTest class as well, please remove the redundant declaration of the same method in that class, and point to the new method that you have added in the utils.